### DISCUSSION TITLE
Missing Scopes

### DISCUSSION THREAD

**Discussion by xplchanx** [Accenture/sfmc-devtools#1885] [General]
Hey there, I'm testing the package for the first time and was thrown the following error for missing scope:

`Missing scope: accounts_read, accounts_write, approvals_read, approvals_write, audiences_read, audiences_write, automations_execute, automations_read, automations_write, calendar_read, calendar_write, campaign_read, campaign_write, contact_bu_mapping_create, contact_bu_mapping_delete, contact_bu_mapping_update, contact_bu_mapping_view, data_extensions_read, data_extensions_write, deep_linking_asset_delete, deep_linking_asset_read, deep_linking_asset_write, deep_linking_settings_read, deep_linking_settings_write, dfu_configure, documents_and_images_write, email_read, email_send, email_write, event_notification_callback_create, event_notification_callback_delete, event_notification_callback_read, event_notification_callback_update, event_notification_subscription_create, event_notification_subscription_delete, event_notification_subscription_read, event_notification_subscription_update, file_locations_read, file_locations_write, journeys_aspr, journeys_delete, journeys_execute, journeys_read, journeys_write, key_manage_revoke, key_manage_rotate, key_manage_view, list_and_subscribers_read, list_and_subscribers_write, marketing_cloud_connect_read, marketing_cloud_connect_send, marketing_cloud_connect_write, ott_channels_read, ott_channels_write, ott_chat_messaging_read, ott_chat_messaging_send, package_manager_deploy, package_manager_package, push_read, push_send, push_write, saved_content_write, sms_read, sms_send, sms_write, social_post, social_publish, social_read, social_write, tags_read, tags_write, tracking_events_read, tracking_events_write, users_read, users_write, web_publish, web_read, web_write, webhooks_read, webhooks_write, workflows_read, workflows_write`

I dont want to provide that level of access to my business unit at this point, is there any way around this? I'm also getting the following `Credentials failure. SOAPError: API Permission Failed.` What am I missing?

**Comment by JoernBerkefeld** [comment]
hi @xplchanx  - by default, mcdev asks to have full access. I never got around to spending time on mapping what exact rights are required for which part of the application. And honestly, it's unlikely I ever will.

In practice, you don't actually need the full scope, but as soon as you get actual **errors**, that's a clear indicator that you are missing something.

I cannot tell you what that Is because you didn't tell me yet what command you were running.
